csp/mod.ts

```diff
@@ -0,0 +1,288 @@
+import type { Middleware, RequestHandler } from "lume/core.ts";
+import { merge } from "lume/core/utils.ts";
+
+const DEFAULT_MAX_AGE = 365 * 86400;
+
+type StrictTransportSecurityOptions = {
+  "max-age": number;
+  "includeSubDomains"?: boolean;
+  "preload"?: boolean;
+};
+
+type ContentSecurityPolicyDirectives =
+  | "child-src"
+  | "connect-src"
+  | "default-src"
+  | "font-src"
+  | "frame-src"
+  | "img-src"
+  | "manifest-src"
+  | "media-src"
+  | "object-src"
+  | "prefetch-src"
+  | "script-src"
+  | "script-src-elem"
+  | "script-src-attr"
+  | "style-src"
+  | "style-src-elem"
+  | "style-src-attr"
+  | "worker-src"
+  | "base-uri"
+  | "plugin-types"
+  | "sandbox"
+  | "form-action"
+  | "frame-ancestors"
+  | "navigate-to"
+  | "report-uri"
+  | "report-to"
+  | "block-all-mixed-content"
+  | "referrer"
+  | "require-sri-for"
+  | "require-trusted-types-for"
+  | "trusted-types"
+  | "upgrade-insecure-requests";
+
+type ReferrerPolicyOptions =
+  | ""
+  | "no-referrer"
+  | "no-referrer-when-downgrade"
+  | "unsafe-url"
+  | "same-origin"
+  | "strict-origin"
+  | "strict-origin-when-cross-origin"
+  | "origin"
+  | "origin-when-cross-origin";
+
+type ExpectCtOptions = {
+  "max-age": number;
+  "enforce"?: boolean;
+  "report-uri"?: string;
+};
+
+type XFrameOptions = "DENY" | "SAMEORIGIN" | boolean | string;
+
+type XPermittedCrossDomainPoliciesOptions =
+  | "none"
+  | "master-only"
+  | "all"
+  | boolean
+  | string;
+
+export interface Options {
+  /** Enforces SSL connections */
+  "Strict-Transport-Security"?: StrictTransportSecurityOptions;
+
+  /** Allows to restrict resources that the web browser loads */
+  "Content-Security-Policy"?: Partial<
+    Record<ContentSecurityPolicyDirectives, string[]>
+  >;
+
+  /** Controls how much referrer information should be included with requests */
+  "Referrer-Policy"?: ReferrerPolicyOptions | ReferrerPolicyOptions[];
+
+  /** Specifies which features of the web browser should be permitted to function */
+  "Permissions-Policy"?: string;
+
+  /** Prevents the use of misissued certificates */
+  "Expect-CT"?: ExpectCtOptions;
+
+  /** Clickjacking protection */
+  "X-Frame-Options"?: XFrameOptions;
+
+  /** MIME sniffing vulnerabilities protection */
+  "X-Content-Type-Options"?: boolean;
+
+  /**  Cross-site scripting (XSS) filter */
+  "X-XSS-Protection"?: boolean;
+
+  /** Restricts loading of Adobe Flash or PDF documents from other domains */
+  "X-Permitted-Cross-Domain-Policies"?: XPermittedCrossDomainPoliciesOptions;
+
+  /** Leaks or fakes information about the server side technology */
+  "X-Powered-By"?: boolean | string;
+}
+
+export const defaults: Options = {
+  "Strict-Transport-Security": {
+    "max-age": DEFAULT_MAX_AGE,
+    "includeSubDomains": true,
+    "preload": true,
+  },
+  "Content-Security-Policy": {
+    "default-src": ["'none'"],
+    "script-src": ["cdn.jsdelivr.net", "'self'"],
+    "style-src": ["cdn.jsdelivr.net", "'self'"],
```

Comment:
I'm not sure about including some domains by default. I think this should be configured by the user, and leave only "self".

Reply:
See comment `All current default values are just for testing purpose with lume.land while dev`